### DISCUSSION TITLE
404 in the folder public

### DISCUSSION THREAD

**Discussion by skrzypczyk** [dunglas/symfony-docker#369] [Q&A]
Hello everybody,
Locally everything works fine, I'm using Encore and everything is fine in my public/build folder. But in my prod (debian), I don't know why I get 404 on everything in the public folder.
Thanks in advance.

**Comment by Omliance** [comment]
Hello everyone, I have the same problem. @skrzypczyk, have you managed to solve it?
It seems that everything is being redirected to the index.php file, even though the app.css file is indeed present in public/build/app.css.

I am using the version php:8.2-fpm-alpine. Everything works in DEV mode with _`docker compose up -d`_, but the problem appears when using _`SERVER_NAME=app.localhost APP_SECRET=xxxxxxxxx CADDY_MERCURE_JWT_SECRET=xxxxxxx docker compose -f docker-compose.yml -f docker-compose.prod.yml up --wait`_.

**Reply by cloclikenweb** [reply]
To address the issue, execute the following command as soon as `docker-compose up` has completed: 
```bash
docker exec container-php-1 tar -C /srv/app/public -cf - . | docker exec -i container-caddy-1 tar -C /srv/app/public -xf -
```
This is necessary because the files in the `public` directory (generated by `npm run build`) of the PHP container are not copied into the `public` directory of the Caddy container during the Docker environment startup process in production.

**Reply by rene-schwabe** [reply]
I have the same issue but there is only a php-container. I don't have a caddy-container. What I´m missing?
@cloclikenweb
